src/overlays/actors/ovl_En_Ending_Hero6/z_en_ending_hero6.c

```diff
@@ -24,20 +55,133 @@ const ActorInit En_Ending_Hero6_InitVars = {
+    Actor_UpdateBgCheckInfo(globalCtx, &this->actor, 20.0f, 20.0f, 50.0f, 0x1DU);
+}
+
+void func_80C23F14(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3s* rot, Actor* thisx) {
+    static Gfx* D_80C2426C[] = { D_060070C0, D_06006FB0, D_06006E80, D_06006D70, D_0600A390 };
+    EnEndingHero6* this = THIS;
+    s32 index;
+
+    OPEN_DISPS(globalCtx->state.gfxCtx);
+
+    if (this->npcIndex >= 4 && limbIndex == 15) {
+        index = this->npcIndex - 4;
+        gSPDisplayList(POLY_OPA_DISP++, D_80C2426C[index]);
+    }
+
+    CLOSE_DISPS(globalCtx->state.gfxCtx);
+}
+
+void EnEndingHero6_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    static TexturePtr D_80C24280[] = { &D_06007350, &D_06009590, &D_06009F90, &D_0600A790, &D_0600AB90 };
+    static TexturePtr D_80C24294[] = { &D_06007750, &D_0600A390, &D_0600A490 };
+    EnEndingHero6* this = THIS;
+    s32 pad;
+    s32 index = 0;
+
+    if (this->unk290 == 1) {
+        OPEN_DISPS(globalCtx->state.gfxCtx);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ending_Hero6/EnEndingHero6_Init.s")
+        func_8012C28C(globalCtx->state.gfxCtx);
+        func_8012C2DC(globalCtx->state.gfxCtx);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ending_Hero6/EnEndingHero6_Destroy.s")
+        if (this->unk284 >= 0) {
+            if (Object_IsLoaded(&globalCtx->objectCtx, this->unk284) != 0) {
+                gSPSegment(POLY_OPA_DISP++, 0x06, globalCtx->objectCtx.status[this->unk284].segment);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ending_Hero6/func_80C23D60.s")
+                switch (this->npcIndex) {
+                    case 4:
+                        gDPSetEnvColor(POLY_OPA_DISP++, 170, 10, 70, 255);
+                        break;
+                    case 5:
+                        gDPSetEnvColor(POLY_OPA_DISP++, 170, 200, 255, 255);
+                        break;
+                    case 6:
+                        gDPSetEnvColor(POLY_OPA_DISP++, 0, 230, 70, 255);
+                        break;
+                    case 7:
+                        gDPSetEnvColor(POLY_OPA_DISP++, 200, 0, 150, 255);
+                        break;
+                    case 8:
+                        gDPSetEnvColor(POLY_OPA_DISP++, 245, 155, 0, 255);
+                        break;
+                }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ending_Hero6/func_80C23DDC.s")
+                if (this->npcIndex == 0) {
+                    gSPSegment(POLY_OPA_DISP++, 0x08, Lib_SegmentedToVirtual(D_80C24280[this->eyeState]));
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ending_Hero6/func_80C23E18.s")
+                    if (this->eyeState < 3) {
+                        index = this->eyeState;
+                    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ending_Hero6/EnEndingHero6_Update.s")
+                    gSPSegment(POLY_OPA_DISP++, 0x09, Lib_SegmentedToVirtual(D_80C24294[index]));
+                }
```

Comment:
This is interesting. These are probably eyes and mouths, although we'd have to look in the objects to find out... I think you said you'd done some work on those already?

Reply:
Indeed I took a quick look at this. I believe that one of these corresponds to the eyes, the other to the eyebrows. I am not 100% sure yet, so I did not want to write anything definitive.

Reply:
Yeah, good idea. Wrong documentation is worse than no documentation.